contracts/CurrencyNetworkRegistry.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity ^0.5.8;
+
+import "./CurrencyNetworkInterface.sol";
+import "./lib/ERC165Query.sol";
+
+
+contract CurrencyNetworkRegistry is ERC165Query {
+
+    struct CurrencyNetworkMetadata {
+        address author;
+        string name;
+        string symbol;
+        uint8 decimals;
+    }
+
+    mapping (address => CurrencyNetworkMetadata) internal currencyNetworks;
+    address[] internal registeredCurrencyNetworks;
+
+    event CurrencyNetworkAdded(
+        address indexed _address,
+        address _author,
+        string _name,
+        string _symbol,
+        uint8 _decimals
+    );
+
+    function addCurrencyNetwork(address _address) external {
+        CurrencyNetworkInterface network;
+
+        require(
+            currencyNetworks[_address].author == address(0) &&
+            bytes(currencyNetworks[_address].name).length == 0 &&
+            bytes(currencyNetworks[_address].symbol).length == 0,
+            "CurrencyNetworks can only be registered once."
+        );
+
+        network = CurrencyNetworkInterface(_address);
+
+        require(
+            this.doesContractImplementInterface(
+                _address,
+                network.transfer.selector ^
+                network.transferFrom.selector ^
+                network.balance.selector ^
+                network.creditline.selector
+            ),
+            "CurrencyNetworks need to implement ERC165 and CurrencyNetworkInterface"
```

Comment:
You should also check that the interface has a `name()`, `symbol()`, and `decimals()` function.

Reply:
Changed the variables into actual methods and extended the interface. See https://github.com/trustlines-protocol/contracts/pull/234/commits/99a70784a4bf5394fc9a0a360629d241e8b5c380

Reply:
The functions you created are automatically created, you didn't need to write them, hence the clashing. 

If you can try removing them again and running the tests, would avoid all this underscore and `_param` names

Reply:
You can not have variables in interfaces, so if we want the methods be part of the interface, we have to add them manually. Also checked back with @schmir-at-bb, this is the best worst case we came up with ;-)